uvtools/plot.py

```diff
@@ -471,6 +472,428 @@ def omni_view_gif(filenames, name='omni_movie.gif'):
+    if np.array(data).dtype != np.dtype('O'):
+        data = np.atleast_2d(data)
+        if not np.iscomplexobj(data):
+            raise TypeError("array-like data must consist of complex numbers.")
+        if data.ndim != 2 or (data.ndim == 2 and 1 in data.shape):
+            raise ValueError("array-like data must be 2-dimensional.")
+    if isinstance(data, np.ndarray):
+        if freqs is None or (times is None and lsts is None):
+            raise ValueError(
+                "freqs and either times or lsts must be provided for plotting an array."
+            )
+        if times is None:
+            time_or_lst = "lst"
+            times = lsts / (2 * np.pi) # For Fourier transform purposes
+        elif lsts is None:
+            time_or_lst = "time"
+    else:
+        try:
+            from pyuvdata import UVData
+            # In case UVData is installed and a non-UVData object was passed.
+            if type(data) is not UVData:
+                raise ImportError
+        except ImportError:
+            raise TypeError("data must either be an ndarray or UVData object.")
+        if antpairpol is None:
+            raise ValueError(
+                "You must provide an antpairpol key if data is a UVData object."
+            )
+        freqs = np.unique(data.freq_array)
+        times = np.unique(data.time_array)
+        lsts = np.unique(data.lst_array)
+        data = data.get_data(antpairpol)
+    
+    # Prepare axis labels.
+    xvals = freqs / 1e6
+    xlabel = "Frequency [MHz]"
+    if time_or_lst == "time":
+        yvals = times - int(times[0])
+        ylabel = f"JD - {int(times[0]):d}"
+    else:
+        yvals = lsts * units.day.to("hr") / (2 * np.pi)
+        ylabel = "LST [hours]"
+
+    # Do any requested Fourier transforms and update axis labels.
+    if fft_axis is not None:
+        freq_taper_kwargs = freq_taper_kwargs or {}
+        time_taper_kwargs = time_taper_kwargs or {}
+        if fft_axis not in ("freq", "time", "both", -1, 0, 1):
+            raise ValueError("fft_axis not recognized.")
+        if type(fft_axis) is int:
+            fft_axis = ("time", "freq", "both")[fft_axis]
+        if fft_axis in ("freq", "both"):
+            delays = utils.fourier_freqs(freqs) * 1e9 # ns
+            data = utils.FFT(data, axis=1, taper=freq_taper, **freq_taper_kwargs)
+            xvals = delays
+            xlabel = "Delay [ns]"
+        if fft_axis in ("time", "both"):
+            fringe_rates = utils.fourier_freqs(times * units.day.to("s")) * 1e3 # mHz
+            data = utils.FFT(data, axis=0, taper=time_taper, **time_taper_kwargs)
+            yvals = fringe_rates
+            ylabel = "Fringe Rate [mHz]"
+
+    # Update data for plotting.
+    data = data_mode(data, mode=mode)
+    vmin, vmax = None, None
+
+    # Prepare colorbar parameters.
+    if mode == "phs":
+        cbar_label = "Phase [radians]"
+        vmin, vmax = -np.pi, np.pi
+        if cmap == "best":
```

Comment:
I've always wanted to find a wrapping colorscale for phase. Every once in a while I check to see if matplotlib or somewhere else has one.  Is Twilight such a thing? If not could you check for me? Someday my dream will come true.

Reply:
"twilight" is indeed a wrapping colorscale. It goes black -> blue -> white -> red -> black (either that, or reversed). There's also "twilight_r", "twilight_shifted" and "hsv". This is a good reference for colormaps: https://matplotlib.org/3.1.1/gallery/color/colormap_reference.html